erpnext/stock/doctype/quality_inspection/quality_inspection.py

```diff
@@ -123,4 +124,23 @@ def postprocess(source, doc):
 		}
 	}, target_doc, postprocess)
 
-	return doc
\ No newline at end of file
+	return doc
+
+@frappe.whitelist()
+def make_quality_inspections(items):
+	items = json.loads(items)
+	quality_inspections = []
+	for item in items:
+		qi = frappe.new_doc("Quality Inspection")
+
+		qi.update({
+			"inspection_type": item.get("inspection_type"),
+			"reference_type": item.get("reference_type"),
+			"reference_name": item.get("reference_name"),
+			"item_code": item.get("item_code"),
+			"sample_size": item.get("sample_size"),
+			"inspected_by": frappe.session.user,
+			"quality_inspection_template": frappe.db.get_value('BOM', item.get("item_code"), 'quality_inspection_template')
+		}).save()
+		quality_inspections.append(qi)
+	return quality_inspections
```

Comment:
Actually, we shouldn't make the QI for the user. Let them create it themselves, just make a mapped doc and return, as done in this file in the `make_quality_inspection` function